### PR TITLE
[Feature] fix #2876, add two fake news datasets

### LINKED ISSUE #2876: Adding new datasets to dgl.data
## 🚀 Feature

Adding a new graph dataset related to node classification (fraud detection) and a new graph dataset related to graph classification (fake news detection) as the default datasets in `dgl.data`. 

## Motivation

[The first graph dataset](https://github.com/YingtongDou/CARE-GNN) includes two homogeneous multi-relational graphs extracted from Yelp and Amazon where nodes represent fraudulent reviews or fraudulent reviewers. It was first proposed in a [CIKM'20 paper](https://arxiv.org/pdf/2008.08692.pdf) and has been used by a recent [WWW'21 paper](https://ponderly.github.io/pub/PCGNN_WWW2021.pdf) as a benchmark. Another [paper](https://arxiv.org/pdf/2104.01404) also takes the dataset as an example to study the non-homophilous graphs. This dataset is built upon industrial data and has rich relational information and unique properties like class-imbalance and feature inconsistency, which makes the dataset be a good instance to investigate how GNNs perform on real-world noisy graphs.

[The second graph dataset](https://github.com/safe-graph/GNN-FakeNews) is composed of two sets of tree-structured fake/real news propagation graphs extracted from Twitter. Different from most of the benchmark datasets for the graph classification task, the graphs in this dataset are tree-structured graphs where the root node represents the news, the leaf nodes are Twitter users who retweeted the root news. Besides, the node features are encoded user historical tweets using different pretrained language models. The dataset could help GNNs learn how to fuse multi-modal information and learn representations for tree-structured graphs. It would be a good addition to current graph classification benchmarks.

## Alternatives

N/A

## Pitch

Adding the above two new datasets as default datasets in `dgl.data`.

## Additional context

N/A

### PR DESCRIPTION
**Pull request by kayzliu**
## Description

Added two fake news datasets, Gossipcop and Politifact. #2876
@zhjwy9343 @classicsong 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/data/fakenews.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    >>> feat = dataset.feature
+    >>> labels = dataset.labels
+    """
+    file_urls = {
+        'gossipcop': 'dataset/FakeNewsGOS.zip',
+        'politifact': 'dataset/FakeNewsPOL.zip'
+    }
+
+    def __init__(self, name, feature_name, raw_dir=None):
+        url = _get_dgl_url(self.file_urls[name])
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply python/dgl/data/fakenews.py:109]: Added based on your comments.

File: python/dgl/data/fakenews.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            bert: 768
+            content: 310
+            profile: 10
+            spacy: 300
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset
+    feature_name : str
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply python/dgl/data/fakenews.py:63]: Modified based on your comments.

File: python/dgl/data/fakenews.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    >>> labels = dataset.labels
+    """
+    file_urls = {
+        'gossipcop': 'dataset/FakeNewsGOS.zip',
+        'politifact': 'dataset/FakeNewsPOL.zip'
+    }
+
+    def __init__(self, name, feature_name, raw_dir=None):
+        url = _get_dgl_url(self.file_urls[name])
+        self.feature_name = feature_name
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply python/dgl/data/fakenews.py:112]: ditto

File: python/dgl/data/fakenews.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kayzliu) [reply python/dgl/data/fakenews.py:110]: Added based on your comments.

File: python/dgl/data/fakenews.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kayzliu) [reply python/dgl/data/fakenews.py:114]: > Raise Error when fail.

Added based on your comments.

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**